src/config.c

```diff
@@ -1891,10 +1941,15 @@ static void enumConfigInit(standardConfig *config) {
 }
 
 static int enumConfigSet(standardConfig *config, sds *argv, int argc, const char **err) {
-    UNUSED(argc);
-    int enumval = configEnumGetValue(config->data.enumd.enum_value, argv[0]);
+    int enumval;
+    if (config->flags & MULTI_ARG_CONFIG) {
+        enumval = configEnumGetValues(config->data.enumd.enum_value, argv, argc);
+    } else {
+        enumval = configEnumGetValue(config->data.enumd.enum_value, argv[0]);
```

Comment:
I'm going to argue against MULTI_ARG_CONFIGs again here. multi-args requires special orchestration to be able to support doing a config set (which only allows a single argument) vs startup/config writing which requires the multiple arguments. I think it makes a lot more sense to have a single quoted string that contains space separate enum values, that we can parse in the configEnumGetValues. I still think multi-args were required purely for compatibility support and should not be readily used for new features.

Introducing a new flag also simplifies the module integration point, since they just can define the flag and not have to deal with the multi_arg_config integration.

Reply:
i didn't suggest the modules deal with multi-arg, i suggested they pass a new `REDISMODULE_CONFIG_BITS` to the existing `RM_RegisterEnumConfig` API.

So from the module's perspective, from the CONFIG command's perspective, and from the config file's perspective, it's the same as what you wrote, right?
i.e. a single argument with spaces, and just a flag for the existing API.

The differences are the way they're passed in command line arguments, and the fact it is handled by the outer config infrastructure, rather than the enums mechanism itself.

Reply:
@oranagra This makes sense to me.